### PR TITLE
Added more information to error when drive cycle starts at t>0

### LINKED ISSUE #3612: More informative error extrapolation drive cycle
At the moment, if a drive cycle starts at `t=1` rather than `t=0` it throws a rather uninformative error (see below)

fe445cd2-3667-45a0-8a78-0807173bf6c3

The easiest way to address this would be:
- Give the interpolant a meaningful name at creation so at least it is clear which interpolant is causing the error https://github.com/pybamm-team/PyBaMM/blob/e0bec2b7c5468e1b448d04bedf42aa8a66d67516/pybamm/experiment/step/_steps_util.py#L123-L125
- (Optionally) Perform a check and if drive cycle starts at `t>0` throw a different, more specific error (or assume it's zero until first provided value, but I personally think this is dangerous).
- Add somewhere in the docs that drive cycles must start at `t=0`, added to #3392 already.

### PR DESCRIPTION
**Pull request by R-Yash**
# Description

When a drive cycle starts at t>0, the current error thrown seems confusing. The changes made gives drive cycle a name and throws relevant exception when t>0. The documentation is changed accordingly to reflect this fact

Fixes #3612 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [x] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [x] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Review by brosaplanella — COMMENTED** [review]
Just a small comment, but looks good overall. Can you also please add a line to the CHANGELOG?

File: pybamm/experiment/step/_steps_util.py [hunk 1]
```diff
@@ -120,8 +120,15 @@ def __init__(
...
+            # Check that drive cycle starts at t=0
+            if t[0] > 0:
+                raise ValueError("Drive cycle must start at t=0")
+
             self.value = pybamm.Interpolant(
-                t, y, pybamm.t - pybamm.InputParameter("start time")
+                t,
+                y,
+                pybamm.t - pybamm.InputParameter("start time"),
+                name="Current Drive Cycle",
```
[the remark this answers is not in the document]
Comment (kratman) [reply pybamm/experiment/step/_steps_util.py:131]:
```suggestion
                name="Drive Cycle",
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Bug Fixes
 
+- updated `_steps_util.py` to throw a specific exception when drive cycle starts at t>0
+
```
Comment (kratman) [inline CHANGELOG.md:6]:
```suggestion
```

**Comment by kratman** [comment]
@R-Yash I hope you don't mind, but I synced with develop so it would be ready for merging

**Comment by R-Yash** [comment]
@kratman Thanks

**Comment by kratman** [comment]
@R-Yash Can you add a test to cover that exception?

**Comment by R-Yash** [comment]
@kratman How do I do that? I am a bit new to testcases.

**Comment by kratman** [comment]
You need to make a test that calls the function where the error is located, and passes in a time that is not zero. You can look at existing tests for that function to see if there is something you can adapt

**Comment by kratman** [comment]
> @kratman 
> Can I modify this existing function ?
> ```
> def test_start_times(self):
>         # Test start_times
>         step = pybamm.step._Step(
>             "current", 1, duration=3600, start_time=datetime(2020, 1, 1, 0, 0, 0)
>         )
>         self.assertEqual(step.start_time, datetime(2020, 1, 1, 0, 0, 0))
> 
>         # Test bad start_times
>         with self.assertRaisesRegex(TypeError, "`start_time` should be"):
>             pybamm.step._Step("current", 1, duration=3600, start_time="bad start_time")

Usually it is good to write a new test. A test like a function should have a single purpose. In this case you just want to test that an error is thrown. It will probably require a "with" statement looking for the thrown exception.

If you are really stuck then we can discuss on slack, but I recommend you give it a shot on your own first

**Comment by R-Yash** [comment]
@kratman I have added a test. Please take a look

**Review by kratman — APPROVED** [review]
Seems good as long as the tests pass

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## Bug Fixes
 
+- updated `_steps_util.py` to throw a specific exception when drive cycle starts at t>0
```
Comment (kratman) [inline CHANGELOG.md:5]:
```suggestion
- Updated `_steps_util.py` to throw a specific exception when drive cycle starts at t>0
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 ## Bug Fixes
 
+- Updated `_steps_util.py` to throw a specific exception when drive cycle starts at t>0
```
Comment (arjxn-py) [inline CHANGELOG.md:5]:
```suggestion
- Updated `_steps_util.py` to throw a specific exception when drive cycle starts at t>0 ([#3756](https://github.com/pybamm-team/PyBaMM/pull/3756))
```
Linking PRs helps us keep track in changelog 🙂

**Review by arjxn-py — APPROVED** [review]
Thanks @R-Yash Looks good, Just a small changelog suggestion.
Also waiting on tests before merging.

File: CHANGELOG.md (hunk 4 above)
Comment (kratman) [reply CHANGELOG.md:5]: @arjxn-py Thanks I missed that when I reviewed

Comment (kratman) [reply CHANGELOG.md]: @R-Yash I merged this so the tests could start

Comment (arjxn-py) [reply CHANGELOG.md]: Thanks @kratman, marking this as resolved
